src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java

```diff
@@ -382,20 +386,37 @@ private CaTMessage convertMessageToCatMessage(
   private CaTMessageOCDS getCaTMessageOCDS(
       final uk.gov.crowncommercial.dts.scale.cat.model.jaggaer.Message message) {
     return new CaTMessageOCDS().date(message.getSendDate()).id(message.getMessageId())
-        .title(message.getSubject()) .author(new CaTMessageOCDSAllOfAuthor()
-                    .id(message.getSender().getId())
-                    .name(message.getSender().getName())
-            );
+        .title(message.getSubject())
+        .author(getAuthorDetails(message));
   }
 
   private MessageOCDS getMessageOCDS(
       final uk.gov.crowncommercial.dts.scale.cat.model.jaggaer.Message message) {
     return new MessageOCDS().date(message.getSendDate()).id(message.getMessageId())
         .title(message.getSubject())
-            .author(new CaTMessageOCDSAllOfAuthor()
-                    .id(message.getSender().getId())
-                    .name(message.getSender().getName())
-            );
+            .description(message.getBody())
+            .author(getAuthorDetails(message));
+  }
+
+  private CaTMessageOCDSAllOfAuthor getAuthorDetails(
+          final uk.gov.crowncommercial.dts.scale.cat.model.jaggaer.Message message) {
+    CaTMessageOCDSAllOfAuthor author;
+    if (CaTMessageNonOCDS.DirectionEnum.RECEIVED.getValue().equals(message.getDirection())) {
+      author = new CaTMessageOCDSAllOfAuthor()
+              .id(message.getSender().getId())
+              .name(message.getSender().getName());
+    } else {
+      var jaggaerUser = userProfileService.resolveBuyerUserByUserId(
+              String.valueOf(message.getSenderUser().getId()))
+              .orElseThrow(() -> new ResourceNotFoundException("Jaggaer"));
```

Comment:
Might want to consider adding the id to the error message to assist any troubleshooting down the line

Reply:
Might also want to apply same to the `Conclave` user error below